azurerm/data_source_application_insights.go

```diff
@@ -0,0 +1,75 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmApplicationInsights() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmApplicationInsightsRead,
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": resourceGroupNameForDataSourceSchema(),
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
```

Comment:
I think using `validate.NoEmptyStrings` here might be better?

Reply:
Makes total sense, changing.